eZ/Publish/Core/FieldType/TextLine/Type.php

```diff
@@ -64,9 +64,9 @@ public function validateValidatorConfiguration($validatorConfiguration)
                 switch ($name) {
                     case 'minStringLength':
                     case 'maxStringLength':
-                        if ($value !== null && !is_integer($value)) {
+                        if ($value !== null && (!is_integer($value) || $value < 0)) {
```

Comment:
We could use two tests, one for integer, one for negative. It would be more readable for both developers and users, imho.